### PR TITLE
Pull out FetchSubmapTexture into cartographer_ros.

### PR DESCRIPTION
**Pull request by SirVer**
This is to make use of it in #432.

### CODE REVIEW

File: cartographer_rviz/cartographer_rviz/drawable_submap.cc [hunk 1]
```diff
@@ -141,7 +140,9 @@ void DrawableSubmap::Update(
 bool DrawableSubmap::MaybeFetchTexture(ros::ServiceClient* const client) {
   ::cartographer::common::MutexLocker locker(&mutex_);
   // Received metadata version can also be lower - if we restarted Cartographer
```
Comment (wohe) [inline cartographer_rviz/cartographer_rviz/drawable_submap.cc]: `Received metadata version can also be lower if we restarted Cartographer.`?

File: cartographer_rviz/cartographer_rviz/drawable_submap.h [hunk 2]
```diff
@@ -99,15 +101,14 @@ class DrawableSubmap : public QObject {
   Ogre::ManualObject* manual_object_;
   Ogre::TexturePtr texture_;
   Ogre::MaterialPtr material_;
-  double submap_z_ = 0. GUARDED_BY(mutex_);
   ::cartographer::transform::Rigid3d pose_ GUARDED_BY(mutex_);
   ::cartographer::transform::Rigid3d slice_pose_ GUARDED_BY(mutex_);
```
Comment (wohe) [inline cartographer_rviz/cartographer_rviz/drawable_submap.h]: Unused? Remove!

File: cartographer_rviz/cartographer_rviz/drawable_submap.cc [hunk 3]
```diff
@@ -153,18 +154,16 @@ bool DrawableSubmap::MaybeFetchTexture(ros::ServiceClient* const client) {
...
   last_query_timestamp_ = now;
   rpc_request_future_ = std::async(std::launch::async, [this, client]() {
-    ::cartographer_ros_msgs::SubmapQuery srv;
-    srv.request.trajectory_id = id_.trajectory_id;
-    srv.request.submap_index = id_.submap_index;
-    if (client->call(srv)) {
+    std::unique_ptr<::cartographer_ros::SubmapTexture> submap_texture =
+        ::cartographer_ros::FetchSubmapTexture(id_, client);
+    ::cartographer::common::MutexLocker locker(&mutex_);
+    query_in_progress_ = false;
```
Comment (wohe) [inline cartographer_rviz/cartographer_rviz/drawable_submap.cc]: Add a comment in the destructor about the logic there: `query_in_progress_` must be true until the Q_EMIT happened. Qt then makes sure that RequestSucceeded() is not executed if this object gets destroyed in the meantime.

**Comment by SirVer** [comment]
@googlebot Wake up, dude?

**Pull request merged**